File: Safe/src/test/java/io/github/cfsima/modernsafe/RestoreDataSetTest.java
package io.github.cfsima.modernsafe;

import org.junit.Test;
import io.github.cfsima.modernsafe.model.RestoreDataSet;

import static org.junit.Assert.assertEquals;

public class RestoreDataSetTest {

    @Test
    public void testSaltAppend() {
        RestoreDataSet dataSet = new RestoreDataSet();
        // We expect setSalt to append.
        // Currently it overwrites, so this test should fail.

        dataSet.setSalt("part1");
        dataSet.setSalt("part2");

        assertEquals("Salt should be appended", "part1part2", dataSet.getSalt());
    }

    @Test
    public void testMasterKeyAppend() {
        RestoreDataSet dataSet = new RestoreDataSet();

        dataSet.setMasterKeyEncrypted("key1");
        dataSet.setMasterKeyEncrypted("key2");

        assertEquals("MasterKey should be appended", "key1key2", dataSet.getMasterKeyEncrypted());
    }

    @Test
    public void testSaltWhitespace() {
        RestoreDataSet dataSet = new RestoreDataSet();
        dataSet.setSalt("\n  someSalt  \n");
        assertEquals("Salt should be trimmed", "someSalt", dataSet.getSalt());
    }

    @Test
    public void testMasterKeyWhitespace() {
        RestoreDataSet dataSet = new RestoreDataSet();
        dataSet.setMasterKeyEncrypted("\n  someKey  \n");
        assertEquals("MasterKeyEncrypted should be trimmed", "someKey", dataSet.getMasterKeyEncrypted());
    }
}

File: Safe/src/test/kotlin/io/github/cfsima/modernsafe/RestoreDataSetTest.kt
package io.github.cfsima.modernsafe

import io.github.cfsima.modernsafe.model.RestoreDataSet
import org.junit.Assert.assertEquals
import org.junit.Test

class RestoreDataSetTest {

    @Test
    fun testSaltAppend() {
        val dataSet = RestoreDataSet()
        // We expect setSalt to append.

        dataSet.salt = "part1"
        dataSet.salt = "part2"

        assertEquals("Salt should be appended", "part1part2", dataSet.salt)
    }

    @Test
    fun testMasterKeyAppend() {
        val dataSet = RestoreDataSet()

        dataSet.masterKeyEncrypted = "key1"
        dataSet.masterKeyEncrypted = "key2"

        assertEquals("MasterKey should be appended", "key1key2", dataSet.masterKeyEncrypted)
    }

    @Test
    fun testSaltWhitespace() {
        val dataSet = RestoreDataSet()
        dataSet.salt = "\n  someSalt  \n"
        assertEquals("Salt should be trimmed", "someSalt", dataSet.salt)
    }

    @Test
    fun testMasterKeyWhitespace() {
        val dataSet = RestoreDataSet()
        dataSet.masterKeyEncrypted = "\n  someKey  \n"
        assertEquals("MasterKeyEncrypted should be trimmed", "someKey", dataSet.masterKeyEncrypted)
    }
}
